src/components/organize/TagsManager/index.tsx

```diff
@@ -0,0 +1,137 @@
+import { useState } from 'react';
+import { Box, Switch, Tooltip, Typography } from '@material-ui/core';
+import { FormattedMessage, useIntl } from 'react-intl';
+
+import ZetkinSection from 'components/ZetkinSection';
+import { ZetkinTag } from 'types/zetkin';
+
+const TagChip: React.FunctionComponent<{ tag: ZetkinTag }> = ({ tag }) => {
+  return (
+    <Tooltip arrow title={tag.description}>
+      <Box
+        bgcolor={tag.color || '#e1e1e1'}
+        borderRadius="18px"
+        fontSize={13}
+        px={2}
+        py={0.7}
+      >
+        {tag.title}
+      </Box>
+    </Tooltip>
+  );
+};
+
+const GroupToggle: React.FunctionComponent<{
+  checked?: boolean;
+  onChange: () => void;
+}> = ({ checked, onChange }) => {
+  const intl = useIntl();
+  return (
+    <Box alignItems="center" display="flex">
+      <Typography variant="body2">
+        {intl.formatMessage({ id: 'misc.tags.tagsManager.groupTags' })}
+      </Typography>
+      <Switch
+        checked={checked}
+        color="primary"
+        data-testid="TagsManager-groupToggle"
+        name="Tags"
+        onChange={onChange}
+      />
+    </Box>
+  );
+};
+
+interface TagsGroups {
+  [key: string]: {
+    tags: ZetkinTag[];
+    title: string;
+  };
+}
+
+const TagsManager: React.FunctionComponent<{
+  appliedTags: ZetkinTag[];
+}> = ({ appliedTags }) => {
+  const intl = useIntl();
+
+  const [isGrouped, setIsGrouped] = useState(false);
+  const groups: TagsGroups = appliedTags.reduce((acc: TagsGroups, tag) => {
+    // Add to ungrouped tags list
+    if (!tag.group) {
+      return {
+        ...acc,
+        ungrouped: {
+          tags: acc['ungrouped'] ? [...acc['ungrouped'].tags, tag] : [tag],
+          title: intl.formatMessage({
+            id: 'misc.tags.tagsManager.ungroupedHeader',
+          }),
+        },
+      };
+    }
+    // Add to tags list for group
+    return {
+      ...acc,
+      [tag.group.id]: {
+        tags: acc[tag.group.id] ? [...acc[tag.group.id].tags, tag] : [tag],
+        title: tag.group.title,
+      },
+    };
+  }, {});
+
+  return (
+    <ZetkinSection
```

Comment:
I agree with @lowlandjuju here, I think. 😊  

I think that a component structure that might make sense at some point is:

* Section (basically what the `TagsManager` is today)
  * Wrapper component (e.g. named `TagsManager`)
    * `TagCloud` (just rendering tags as either grouped or not)
    * `TagSelect` (to search, select, add, create etc)

The full section could be used here in the person profile page. The wrapper only could be used in the journeys screen (#576) and the `TagCloud` could be used in things like the hover card (#614).

But that's refactoring that can happen as it's needed, so I also don't think the merge needs to be blocked by this.

Reply:
I felt like `grouped` made more sense an an internal thing and not depend on an external state. And there isn't really a case where the `TagManager` doesn't have a title associated with it. So I feel like grouping everything together is fine because it's the use case we have. And because the toggle goes in the section component, I think it makes sense to just include everything together.

Regarding the components internally Richard: yes you're right as complexity grows I'm going to need to split things out. Will do that as I go.